source/GameInterface/Services/ObjectManager/ObjectManager.cs

```diff
@@ -45,13 +44,21 @@ public interface IObjectManager
     bool TryGetId(object obj, out string id);
 
     /// <summary>
-    /// Attempts to get an object using a StringId and object type
+    /// Attempts to get a <see cref="MBObjectBase"/>-based object using a StringId and object type
     /// </summary>
     /// <typeparam name="T">Type of object</typeparam>
     /// <param name="id">StringId used to lookup object</param>
     /// <param name="obj">Out parameter for the object (null if not found)</param>
     /// <returns>True if successful, false if failed</returns>
-    bool TryGetObject<T>(string id, out T obj) where T : class;
+    bool TryGetObject<T>(string id, out T obj) where T : MBObjectBase;
+    /// <summary>
+    /// Attempts to get an object not based on <see cref="MBObjectBase"/> object using a StringId and object type
+    /// </summary>
+    /// <typeparam name="T">Type of object</typeparam>
+    /// <param name="id">StringId used to lookup object</param>
+    /// <param name="obj">Out parameter for the object (null if not found)</param>
+    /// <returns>True if successful, false if failed</returns>
+    bool TryGetNonMBObject<T>(string id, out T obj) where T : class;
```

Comment:
Revert this change please. `bool TryGetObject<T>(string id, out T obj) where T : class;` should only exist